scripts/build-install.sh

```diff
@@ -22,7 +22,7 @@ if [[ -n "$(which ninja)" ]]; then cmake_args+=" -GNinja"; fi
 cmake_args+="
 -D Legion_USE_CUDA=ON
 -D Legion_USE_OpenMP=ON
--D CMAKE_CUDA_ARCHITECTURES=NATIVE
+-D Legion_CUDA_ARCH=native
```

Comment:
Are we encouraging users not to use `CMAKE_CUDA_ARCHITECTURES`?

Reply:
`CMAKE_CUDA_ARCHITECTURES` applies to all CUDA targets in the configuration, but I've made Legion [set the `CUDA_ARCHITECTURES` property](https://gitlab.com/StanfordLegion/legion/-/merge_requests/739/diffs#6d9c9297ab43e0b6df8fe6898c740e1683f74b6b_507_503) on each target, then [expose](https://gitlab.com/StanfordLegion/legion/-/merge_requests/739/diffs#eb3b097f0162225e2fcae76f74c070e17c5552c2_37_28) its archs in its package config via `Legion_CUDA_ARCH`.

I updated legate.core to inherit the archs list that Legion uses (similar to what we do for `Legion_USE_CUDA` etc.), so that way we don't get into a situation where legate.core is compiled for an arch not supported by Legion.

The two reasons Legion is setting the `CUDA_ARCHITECTURES` property per target instead of setting `CMAKE_CUDA_ARCHITECTURES` globally is:
1. This prevents Legion's archs list doesn't leak out of Legion's CMakeLists.txt and affect unrelated targets. If someone is building Legion as a submodule (like we do in legate.core), setting `CMAKE_CUDA_ARCHITECTURES` globally can affect the rest of their build.
3. Legion includes a polyfill for `CUDA_ARCHITECTURES` that translates each arch to its equivalent `-gencode` flag in < CMake 3.18. We can either append the `-gencode` flags to the `CMAKE_CUDA_FLAGS` global or set them per-target via `target_compile_options()`, and the strong preference is to avoid touching the global.